src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -2600,5 +2623,273 @@ private void Remove(int startIndex, int count, out StringBuilder chunk, out int
+                // formatting into temporary space and then appending that result along with any required padding.
+                AppendFormattedWithTempSpace(value, alignment, format: null);
+
+            /// <summary>Writes the specified value to the builder.</summary>
+            /// <param name="value">The value to write.</param>
+            /// <param name="format">The format string.</param>
+            /// <param name="alignment">Minimum number of characters that should be written for this value.  If the value is negative, it indicates left-aligned and the required minimum is the absolute value.</param>
+            public void AppendFormatted<T>(T value, int alignment, string? format)
+            {
+                if (alignment == 0)
+                {
+                    // This overload is used as a fallback from several disambiguation overloads, so special-case 0
+                    // to avoid using temporary space if we don't have to.
+                    AppendFormatted(value, format);
+                }
+                else
+                {
+                    AppendFormattedWithTempSpace(value, alignment, format);
+                }
+            }
+
+            private void AppendFormattedWithTempSpace<T>(T value, int alignment, string? format)
+            {
+                // It's expected that either there's not enough space in the current chunk to store this formatted value,
+                // or we have a non-0 alignment that could require padding inserted. So format into temporary space and
+                // then append that written span into the StringBuilder: StringBuilder.Append(span) is able to split the
+                // span across the current chunk and any additional chunks required.
+
+                var builder = new InterpolatedStringBuilder(_provider, stackalloc char[256]);
+                builder.AppendFormatted(value, format);
+                AppendFormatted(builder.Text, alignment);
+                builder.Clear();
+            }
+            #endregion
+
+            #region AppendFormatted ReadOnlySpan<char>
+            /// <summary>Writes the specified character span to the builder.</summary>
+            /// <param name="value">The span to write.</param>
+            public void AppendFormatted(ReadOnlySpan<char> value) => _stringBuilder.Append(value);
+
+            /// <summary>Writes the specified string of chars to the builder.</summary>
+            /// <param name="value">The span to write.</param>
+            /// <param name="alignment">Minimum number of characters that should be written for this value.  If the value is negative, it indicates left-aligned and the required minimum is the absolute value.</param>
+            /// <param name="format">The format string.</param>
+            public void AppendFormatted(ReadOnlySpan<char> value, int alignment = 0, string? format = null)
```

Comment:
I assume `format` is an ignored argument because

a) The pattern requires it.
b) `string.Format("{0:X}", "hi");` doesn't complain that format specifiers make no sense in context, so we won't here?

Reply:
(b)